docs/src/interface-definition.md

```diff
@@ -20,6 +20,31 @@ You do not need to implement all of these by yourself. There is macro that will
 
 Each of the individual calls have two common inputs: `AtomsBase.AbstractSystem` compatible structure and a `calculator` that incudes details of the calculation method. Calculate interface has additionally the type of calculation as the first input. You can tune calculation by passing keyword arguments, which can be ignored, but they need to be present in the function definition.
 
+## High-level vs Low-level Interface
+The philosophy behing AtomsCalculators.jl is to provide high-level calls with default 
+parameters through `potential_energy`, `forces`, `forces!` and `virial` and low-level calls 
+with user-specifiable parameters through `calculate`. 
+
+The low-level `calculate` interface follows the [Lux](https://lux.csail.mit.edu/stable/) model for parameters and state. This means that calculators are **immutable** structs that are passed 
```

Comment:
calculators will NOT be immutable. But they need to ACT as immutable. This is a subtle but important difference.